system/sha3/wrapper/src/lib.rs

```diff
@@ -0,0 +1,88 @@
+pub mod wrap;
+pub use wrap::*;
+use sha3;
+use sha3::{Digest, digest::{Update, ExtendableOutput, XofReader}};
+use hex;
+
+pub fn sha3_512(args: ArgsSha3512) -> String {
+    let data = sha3::Sha3_512::new().chain_update(args.message).finalize();
+    return hex::encode(data);
+}
+
+pub fn sha3_384(args: ArgsSha3384) -> String {
+    let data = sha3::Sha3_384::new().chain_update(args.message).finalize();
+    return hex::encode(data);
+}
+
+pub fn sha3_256(args: ArgsSha3256) -> String {
+    let data = sha3::Sha3_256::new().chain_update(args.message).finalize();
+    return hex::encode(data);
+}
+
+pub fn sha3_224(args: ArgsSha3224) -> String {
+    let data = sha3::Sha3_224::new().chain_update(args.message).finalize();
+    return hex::encode(data);
+}
+
+pub fn keccak_512(args: ArgsKeccak512) -> String {
+    let data = sha3::Keccak512::new().chain_update(args.message).finalize();
+    return hex::encode(data);
+}
+
+pub fn keccak_384(args: ArgsKeccak384) -> String {
+    let data = sha3::Keccak384::new().chain_update(args.message).finalize();
+    return hex::encode(data);
+}
+
+pub fn keccak_256(args: ArgsKeccak256) -> String {
+    let data = sha3::Keccak256::new().chain_update(args.message).finalize();
+    return hex::encode(data);
+}
+
+pub fn keccak_224(args: ArgsKeccak224) -> String {
+    let data = sha3::Keccak224::new().chain_update(args.message).finalize();
```

Comment:
Why aren't you borrowing `args.message` here?
`let data = sha3::Keccak224::new().chain_update(args.message).finalize();`

Reply:
I fixed this. Thanks for catching it!